subsys/bluetooth/host/conn.c

```diff
@@ -1114,13 +1114,27 @@ static void add_pending_tx(struct bt_conn *conn, bt_conn_tx_cb_t cb)
 	key = irq_lock();
 	sys_slist_append(&conn->tx_pending, node);
 	irq_unlock(key);
+
+	return node;
+}
+
+static void remove_pending_tx(struct bt_conn *conn, sys_snode_t *node)
+{
+	unsigned int key;
+
+	key = irq_lock();
```

Comment:
It would be a good idea to start using semaphores with init value of 1 or mutexes instead of disabling interrupts. Disabling interrupts can have serious consequences with the real-time guarantees of the controller. Also, isn't this managed by a queue and a tx thread? why do you need to lock at all?